Src/IronPython/Compiler/Ast/ClassDefinition.cs

```diff
@@ -194,11 +208,17 @@ public class ClassDefinition : ScopeStatement {
             var createLocal = CreateLocalContext(_parentContextParam);
 
             init.Add(Ast.Assign(LocalCodeContextVariable, createLocal));
+            // __classcell__ == ClosureCell(__class__)
+            if (needClassCell) {
+                MSAst.Expression assignClassCell = AssignValue(GetVariableExpression(ClassCellVariable), GetVariableExpression(ClassVariable)) ;
```

Comment:
@BCSharp here I need your help, the plan is to do:
```python
__classcell__ = ClosureCell(__class__)
```

but it turns into:

```python
__classcell__ = __class__
```

is there anyway to get the `ClosureCell` object from the context?

Reply:
Initializing `__class__` during `__new__` is an interesting approach, I haven't tried it. 

I don't think you can simply do `__classcell__ = ClosureCell(__class__)` at this point, but you can set up a closure for holding `__class__` upfront, this is what I did in 2d95476.